Celeste.Mod.mm/Mod/Entities/EntityCollider.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Monocle;
+using System;
+using Microsoft.Xna.Framework;
+
+namespace Celeste.Mod.Entities {
+    [Tracked(false)]
+    public class EntityCollider<T> : Component where T : Entity {
+        public readonly string entityType = typeof(T).Name;
+
+        public Action<T> OnEntityAction;
+
+        public Collider Collider;
+
+        public EntityCollider(Action<T> onEntityAction, Collider collider)
+            : base(active: true, visible: true) {
+            OnEntityAction = onEntityAction;
+            Collider = collider;
+        }
+
+        public override void Added(Entity entity) {
+            if (!Engine.Scene.Tracker.IsEntityTracked<T>()) {
```

Comment:
Prefer not using `Engine.Scene` in a context where you are in a scene already.
Rather use `entity.Scene` inside `EntityAdded`.